### DISCUSSION TITLE
Amazon S3 Delivery with Java SDK

### DISCUSSION THREAD

**Discussion by wolffbruno** [awslabs/amazon-kinesis-video-streams-producer-sdk-java#187] [Q&A] [answered]
According to the AWS documentation, to activate image generation for my stream, I need to set a flag in the metadata directly on the producer side. How can I do this using the Java SDK?

https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/gs-s3Delivery.html

**Comment by sirknightj** [comment, marked as answer]
Apologies for the late response

This feature has only been recently been ported over to the Java in this PR: https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java/pull/283 (1.14.0 release)

To every fragment you want to tag, call this function once.

```java
kinesisVideoStream.putEventMetadata(StreamEventType.STREAM_EVENT_TYPE_IMAGE_GENERATION.getIntType(), null);
```
